FileShareClient/FileShareApiClient.cs

```diff
@@ -106,6 +112,51 @@ public async Task<Stream> DownloadFileAsync(string batchId, string filename)
             }
         }
 
+        public async Task<IResult<DownloadFileResponse>> DownloadFileAsync(string batchId, string fileName, FileStream destinationStream, long fileSizeInBytes = 0, CancellationToken cancellationToken = default)
```

Comment:
destinationStream should be Stream not FileStream. There is no reason to restrict the destination to a file. The user of the library might want to stream directly to a MemoryStream, a compressed stream, a network socket, or a zip archive for example.
```suggestion
        public async Task<IResult<DownloadFileResponse>> DownloadFileAsync(string batchId, string fileName, Stream destinationStream, long fileSizeInBytes = 0, CancellationToken cancellationToken = default)
```

Reply:
agree, updated as Stream. Thanks